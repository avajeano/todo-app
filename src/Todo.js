import React from "react";

const Todo = ({id, todo, removeTodo }) => {
    const handleRemove = () => {
        removeTodo(id);
    };
    
    return (
        <ul>
            <li>{todo}
            <button onClick={handleRemove}>X</button>
            </li>
        </ul>
    )
}


export default Todo;