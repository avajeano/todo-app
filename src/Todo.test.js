import React from "react";
import { render } from '@testing-library/react';
import Todo from "./Todo";

// Todo smoke test
it('renders without crashing', () => {
    render(<Todo id="test-id" removeTodo={() => {}}/>)
})

// Todo snapshot test
it('matches snapshot', () => {
    const { asFragment } = render(<Todo id="test-id" removeTodo={() => {}}/>)
    expect(asFragment()).toMatchSnapshot()
})