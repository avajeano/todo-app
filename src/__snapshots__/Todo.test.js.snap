// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <button>
        X
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
